zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -451,46 +491,44 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
         };
 
         self.tx_id
-            .assign(region, offset, Value::known(F::from(tx.id as u64)))?;
+            .assign(region, offset, Some(tx.id.to_le_bytes()))?;
         self.tx_nonce
             .assign(region, offset, Value::known(F::from(tx.nonce)))?;
         self.tx_gas
             .assign(region, offset, Value::known(F::from(tx.gas)))?;
         self.tx_gas_price
-            .assign(region, offset, Some(tx.gas_price.to_le_bytes()))?;
+            .assign_u256(region, offset, tx.gas_price)?;
         self.mul_gas_fee_by_gas
             .assign(region, offset, tx.gas_price, tx.gas, gas_fee)?;
-        let caller_address = tx
-            .caller_address
-            .to_scalar()
-            .expect("unexpected Address -> Scalar conversion failure");
-        let callee_address = tx
-            .callee_address
-            .to_scalar()
-            .expect("unexpected Address -> Scalar conversion failure");
         self.tx_caller_address
-            .assign(region, offset, Value::known(caller_address))?;
-        self.tx_caller_address_is_zero
-            .assign(region, offset, caller_address)?;
+            .assign_h160(region, offset, tx.caller_address)?;
+        self.tx_caller_address_is_zero.assign_u256(
+            region,
+            offset,
+            U256::from_big_endian(&tx.caller_address.to_fixed_bytes()),
+        )?;
         self.tx_callee_address
-            .assign(region, offset, Value::known(callee_address))?;
+            .assign_h160(region, offset, tx.callee_address)?;
         self.call_callee_address.assign(
             region,
             offset,
-            Value::known(
-                if tx.is_create {
-                    get_contract_address(tx.caller_address, tx.nonce)
-                } else {
-                    tx.callee_address
-                }
-                .to_scalar()
-                .expect("unexpected Address -> Scalar conversion failure"),
-            ),
+            if tx.is_create {
+                get_contract_address(tx.caller_address, tx.nonce)
+            } else {
+                tx.callee_address
+            }
+            .to_fixed_bytes()
+            .iter()
+            .rev()
+            .copied()
+            .collect_vec()
+            .try_into()
+            .ok(),
```

Comment:
wow!  Hopefully this can be improvement following my previous comment and having `.assign_h160`